src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -434,8 +434,14 @@ def _to_data_or_iter(
         results_cursor: SnowflakeCursor,
         to_pandas: bool = False,
         to_iter: bool = False,
-        num_statements: Optional[int] = None,
     ) -> Dict[str, Any]:
+        if to_iter and not to_pandas:  # Fix for SNOW-869536
```

Comment:
`to_pandas` doesn't have this issue, `SnowflakeCursor.fetch_pandas_batches` already handles the isolation.

Reply:
Can we put this into comment instead? This seems more informative than "fix for" :)